samples/net/sockets/echo_client/boards/intel_socfpga_agilex5_socdk.overlay

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* The overlay file should be used to enable any
+ * dts nodes required by this application for this
+ * board.
+ */
+
+&xgmac0 {
+	status = "okay";
+	local-mac-address = [06 00 00 00 00 01];
```

Comment:
Why do we hard code the mac address, shouldn't this come from device itself and if not, shouldn't we then use a randomized value in order to avoid conflicts in the network?

Reply:
Hi, @jukkar
For the first point, a default MAC address is set in the device dtsi file for each XGMAC device node. Nevertheless, I have manually assigned distinct MAC addresses for the client and server applications in their corresponding board overlay files. This ensures that these two applications are compiled with different MAC addresses, preventing any conflict.

For the second, I guess to randomize the MAC address we need a True Random Number Generator(TRNG) but the the target platform does not have TRNG. Please correct me if I am wrong.

This driver implements a set_config API to update the MAC address during the runtime.

Reply:
Using static mac address is still not good as then it prevents two boards having the same software to be run same time in the network. You can certainly support static MAC address in the driver, but I would suggest you do not hard code these in the samples. User can shoot themselves to the foot if they want to but we do not need to promote this.

Please check for example `driver/ethernet/eth_stm32_hal.c` and search `zephyr_random_mac_address` to see how to implement random mac address support. Other Ethernet drivers are doing same things, the stm32 driver was the first one I looked. I would not worry about TRNG, if it is not available then let the system use what is available. The MAC address does not need to be selected securely, it is enough that it is random enough to avoid collisions.

Reply:
Implemented random mac address support